src/main/java/org/breedinginsight/daos/ProgramUploadDAO.java

```diff
@@ -0,0 +1,112 @@
+/*
+ * See the NOTICE file distributed with this work for additional information
+ * regarding copyright ownership.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.breedinginsight.daos;
+
+import org.breedinginsight.dao.db.enums.UploadType;
+import org.breedinginsight.dao.db.tables.BiUserTable;
+import org.breedinginsight.dao.db.tables.daos.BatchUploadDao;
+import org.breedinginsight.model.*;
+import org.breedinginsight.model.User;
+import org.jooq.*;
+
+import javax.inject.Inject;
+import javax.inject.Singleton;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import java.util.UUID;
+
+import static org.breedinginsight.dao.db.Tables.*;
+
+@Singleton
+public class ProgramUploadDAO extends BatchUploadDao {
+
+    private DSLContext dsl;
+
+    @Inject
+    public ProgramUploadDAO(Configuration config, DSLContext dsl) {
+        super(config);
+        this.dsl = dsl;
+    }
+
+    public List<ProgramUpload> getUploads(UUID programId, UUID userId, UploadType type) {
+
+        List<Record> records = getUploadsQuery()
+                .where(BATCH_UPLOAD.PROGRAM_ID.eq(programId)
+                        .and(BATCH_UPLOAD.USER_ID.eq(userId))
+                        .and(BATCH_UPLOAD.TYPE.eq(type)))
+                .fetch();
+
+        return parseRecords(records);
+
+    }
```

Comment:
@ctucker3 

>if our enums change in the future, we can change them in the dao instead of all the uses of the dao

This would apply to renaming of an enum, not necessarily adding an enum, right?

To your point of having the enum in the code vs in the db, I can see it going either way, just my 2 (not helpful) cents :)

Reply:
Ya, I considered that and just ended up following the table structure. It would probably make things a bit cleaner as you described to have different methods for each type, will just have to add them each time we do a new upload type rather than just using what's there.

Reply:
Talked with @timparsons, planning to leave the DAO as-is where services will pass in the upload type. The enum will also remain in the db. You ok with that @ctucker3?

Reply:
Fine with me!